### ISSUE TITLE
Empty editor - sets form control value to null

### ISSUE THREAD

**Issue by Y4Rick** [KillerCodeMonkey/ngx-quill#1257]
Initial form control value is empty string = message: [''].
But when you tap smth and remove all in editor - the control sets to null not to empty string as was from the beginning.

And there's no way to get `this` control reference - for setting it to empty string manually!
For example: get control reference in customModule constructor - it would be a good thing!

**Comment by KillerCodeMonkey** [comment]
i do not get what you mean with "to get this control reference". what do you mean with control and reference?

if you declared you own form control you can just set the value when and if you like?

**Comment by Y4Rick** [comment]
Yes - I can listen to valueChanges for particular control - and reset value any - what I want. 
But if I have 10 controls (10 quill editors) - i should to do this for each control - it's unuseful code! Even if listen to whole form - still unuseful!
And I what to do this in one place - for any editor which appears in future.

I create custom module - which applied in `foRoot` config globally
```
  customModules: [
    {
      implementation: myModule,
      path: 'modules/myModule',
    },
```
Inside this module I can subscribe to event `text-change` ` this.quill.on('text-change'...`
Where I can check if the editor is empty `this.quill.getText().length <= 1` and than set control value = to what I want! But I get no control reference here!
Even if I `setText('')` the value still `null`

**Comment by KillerCodeMonkey** [comment]
I do Not get what you wanna do with The custom Module either. Those are
native quilljs Modules... They know nothing about angular

You could create a custom angular component wrapping quill-editor
Component. Or Just send a PR where you can Just Pass The emptyValue of the
Editor (null or empty String)


Am 26.05.2021 in 16:38, YaRick Krasatyuk ***@***.***>
schrieb:

Yes - I can listen to valueChanges for particular control - and reset value
any - what I want.
But if I have 10 controls (10 quill editors) - i should to do this for each
control - it's unuseful code! Even if listen to whole form - still unuseful!
And I what to do this in one place - for any editor which appears in future.

I create custom module - which applied in foRoot config globally

  customModules: [
    {
      implementation: myModule,
      path: 'modules/myModule',
    },

Inside this module I can subscribe to event text-change
this.quill.on('text-change'...
Where I can check if the editor is empty this.quill.getText().length <= 1
and than set control value = to what I want! But I get no control reference
here!
Even if I setText('') the value still null

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/KillerCodeMonkey/ngx-quill/issues/1257#issuecomment-848824848>,
or unsubscribe

**Comment by Y4Rick** [comment]
ok - quill editor works with formControl/formControlName - tit means that editor have `valueAccessor`
can I get access to it?

**Comment by Y4Rick** [comment]
quill editor - is TEXT editor - it should not set value to null 
```
    if (html === '&lt;p&gt;&lt;br&gt;&lt;/p&gt;' || html === '&lt;div&gt;&lt;br&gt;&lt;/div&gt;') {      html = null    }```

**Comment by KillerCodeMonkey** [comment]
Like i Said. Just send me a PR ;-)


Am 26.05.2021 in 17:07, YaRick Krasatyuk ***@***.***>
schrieb:

quill editor - is TEXT editor - it should not set value to null

    if (html === '&lt;p&gt;&lt;br&gt;&lt;/p&gt;' || html ===
'&lt;div&gt;&lt;br&gt;&lt;/div&gt;') {      html = null    }```

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/KillerCodeMonkey/ngx-quill/issues/1257#issuecomment-848849453>,
or unsubscribe

**Comment by Y4Rick** [comment]
ok - in progress...

**Comment by KillerCodeMonkey** [comment]
Would be nice to have an Input and global config setting to switch between
null and empty String.

Null would be The Default to Not Introduce a breaking Change for Others :-).

Thanks


Am 26.05.2021 in 17:12, YaRick Krasatyuk ***@***.***>
schrieb:

ok - in progress...

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/KillerCodeMonkey/ngx-quill/issues/1257#issuecomment-848854097>,
or unsubscribe

**Comment by Y4Rick** [comment]
the first result is that global config did't set
 when using this like input on element
 ```
        <quill-editor
          defaultEmptyValue=""
          formControlName="message"
        ></quill-editor>
```
it works fine

but globally config did't set 
```
export const QUILL_CONFIG: QuillConfig = {
  customModules: [
    {
      implementation: ImageReformer,
      path: 'modules/imageReformer',
    },
  ],
  modules: {
    toolbar: [
      ['bold', 'italic', 'underline'],
      [{ list: 'ordered' }, { list: 'bullet' }],
      [{ indent: '-1' }, { indent: '+1' }],
      ['code-block', 'link', 'image'],
      ['clean'],
    ],
    imageReformer: true,
  },
  bounds: 'self',
  defaultEmptyValue: 'q',
};
```
in debugger i see my options in quill constructor 
![Снимок экрана 2021-05-26 в 19 50 31](https://user-images.githubusercontent.com/39103122/119701215-c79a9f00-be5c-11eb-8e78-fe17797c2e87.png)

but after init quill options do not have my options
![Снимок экрана 2021-05-26 в 19 53 00](https://user-images.githubusercontent.com/39103122/119701406-f57fe380-be5c-11eb-8360-15a4612a4add.png)

what is the problem ?

**Comment by KillerCodeMonkey** [comment]
because QuillConfig (in quillservice) is something of ngx-quill and you are just inspecting the quilljs instance ;-).. so you will not find it there

**Comment by Y4Rick** [comment]
ready
but ERROR: Permission to KillerCodeMonkey/ngx-quill.git denied to Y4Rick.

**Comment by KillerCodeMonkey** [comment]
i do not know how much you now about git ... you need to fork the repo and create a pull request from that forked repo..

**Comment by Y4Rick** [comment]
1. git clone https://github.com/KillerCodeMonkey/ngx-quill.git
2. git checkout -b my branch
3. git commit
4. git push origin default-empty-value-setter  

ERROR: Permission to KillerCodeMonkey/ngx-quill.git denied to Y4Rick.

**Comment by Y4Rick** [comment]
hm... user from gitlab not from github

**Comment by KillerCodeMonkey** [comment]
dude... i said fork not clone.. so you only make changes in a repo you own.
and then you send a pull request to my repo...

Am Mi., 26. Mai 2021 um 22:27 Uhr schrieb YaRick Krasatyuk <
***@***.***>:

**Comment by Y4Rick** [comment]
finally done
check please https://github.com/KillerCodeMonkey/ngx-quill/pull/1260

**Comment by Y4Rick** [comment]
updated
lint fix
https://github.com/KillerCodeMonkey/ngx-quill/pull/1260

**Comment by Y4Rick** [comment]
I don't want to be too forward.
Any update to this https://github.com/KillerCodeMonkey/ngx-quill/pull/1260

**Comment by KillerCodeMonkey** [comment]
sorry i forgot to publish my review comment *facepalm

**Comment by KillerCodeMonkey** [comment]
are you back? can we finish this baby?

**Comment by Y4Rick** [comment]
updated 
not sure that the test is correct
it is me first test )

**Comment by KillerCodeMonkey** [comment]
released in v14.1.0. Thanks again :)
